cypress/e2e/EngageSphere_GUI.cy.js

```diff
@@ -0,0 +1,57 @@
+// Renderiza o cabeçalho com um h1 e alternância de tema
+describe('Teste do cabeçalho e alternância de tema', () => {
+    beforeEach(() => {
+      cy.visit('/');
+    });
+
+    it('Deve renderizar o cabeçalho com um h1', () => {
+      cy.contains('h1', 'EngageSphere').should('be.visible');
+    });
+
+    it('Deve alternar o tema', () => {
+      cy.get('#theme-toggle-button').should('exist');
```

Comment:
Que tal verificar a visiblidade do elemento, em vez de só sua existência?

```suggestion
      cy.get('#theme-toggle-button').should('be.visible');
```